internal/plugin/plugin.go

```diff
@@ -342,9 +344,38 @@ func calculateGitModified(ctx context.Context, pluginYamlPath string) (bool, err
 	return strings.TrimSpace(output.String()) != "", nil
 }
 
-// changedFiles contains data from the tj-actions/changed-files action.
-// See https://github.com/tj-actions/changed-files#outputs for more details.
-type changedFiles struct {
-	AnyModified      string   `env:"ANY_MODIFIED"`
-	AllModifiedFiles []string `env:"ALL_MODIFIED_FILES"`
+func getBaseRefFromEnv(ctx context.Context, lookuper envconfig.Lookuper) (string, error) {
+	type env struct {
+		BaseRef string `env:"BASE_REF"`
+	}
+	var e env
+	if err := envconfig.ProcessWith(ctx, &envconfig.Config{
+		Target:   &e,
+		Lookuper: lookuper,
+	}); err != nil {
+		return "", fmt.Errorf("envconfig process with: %w", err)
+	}
+	if e.BaseRef == "" {
+		return "", fmt.Errorf("empty or missing BASE_REF")
+	}
+	return e.BaseRef, nil
+}
+
+// filterPluginPaths returns the filepaths that are considered to be a relevant plugin file.
+func filterPluginPaths(filePaths []string) []string {
+	return slicesext.Filter(
+		filePaths,
+		func(filePath string) bool {
+			// Include all files within plugins/*
+			if !strings.HasPrefix(filePath, "plugins/") {
+				return false
+			}
+			// We only care about files in the plugins version dirs, no deeper, which means exactly 5
+			// parts in the path:
+			//   plugins/<owner_name>/<plugin_name>/<version>/*
+			//  | 1     | 2          | 3           | 4       | 5
+			parts := strings.Split(filePath, "/")
+			return len(parts) == 5
```

Comment:
Just in case any of the docker context ever comes from subdirs:

```suggestion
			return len(parts) >= 5
```

Reply:
Ah ok, so anything within the versions dir, even in subdirs?